lib/armv6m-atomic-hack/src/lib.rs

```diff
@@ -0,0 +1,94 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Provides fake atomic read-modify-write operations for situations where you
+//! _really_ know what you're doing.
+//!
+//! Pulling this trait in can cause code written for ARMv7-M and later machines,
+//! which have atomic read-modify-write operations, to compile on ARMv6-M. This
+//! is, in general, not safe: the program wanted an atomic read-modify-write and
+//! you're faking it with a non-atomic sequence. However, in our _specific_ case
+//! on Hubris, we can do this safely because
+//!
+//! 1. Tasks are isolated.
+//! 2. Tasks are single-threaded.
+//! 3. ISRs do not access memory shared with tasks.
+//!
+//! If any of those three points is wrong in your case, do not use these, it
+//! will go badly for you.
+//!
+//! Everything in this crate is conditional on the `armv6m` config, which means
+//! (1) it will only work inside the Hubris build system and (2) accidentally
+//! including it on armv7m or later won't pull in the bogus implementations.
+//!
+//! # Why not just disable interrupts
+//!
+//! Because (1) we can't from unprivileged mode and (2) we don't out of
+//! principle anyway.
+
+#![no_std]
+
+use core::sync::atomic::{AtomicBool, AtomicU32, Ordering};
```

Comment:
You probably want `#[cfg(armv6m)]` here too so that Github Actions doesn't get _really mad_ (or wrap the whole file in that conditional?)

Reply:
Ha, good point. Wrapping the whole file in the conditional made things unhappy, iirc.